executor/joiner.go

```diff
@@ -400,26 +365,26 @@ func (j *innerJoiner) tryToMatch(outer chunk.Row, inners chunk.Iterator, chk *ch
 	if inners.Len() == 0 {
 		return false, nil
 	}
-	j.chk.Reset()
-	chkForJoin := j.chk
-	if len(j.conditions) == 0 {
-		chkForJoin = chk
-	}
-	inner, numToAppend := inners.Current(), j.maxChunkSize-chk.NumRows()
-	for ; inner != inners.End() && numToAppend > 0; inner, numToAppend = inners.Next(), numToAppend-1 {
-		if j.outerIsRight {
-			j.makeJoinRowToChunk(chkForJoin, inner, outer)
-		} else {
-			j.makeJoinRowToChunk(chkForJoin, outer, inner)
+
+	numToAppend := j.maxChunkSize - chk.NumRows()
+	return tryToMatchInerAndOuter(j.ctx, j.outerIsRight, outer, inners, j.conditions, j.mutRow, chk, numToAppend)
+}
+
+func tryToMatchInerAndOuter(ctx sessionctx.Context, isRight bool, outer chunk.Row, inners chunk.Iterator, conditions []expression.Expression, mutRow chunk.MutRow, outChk *chunk.Chunk, numToAppend int) (bool, error) {
```

Comment:
1.misspelling
s/ tryToMatchInerAndOuter/ tryToMatchInnerAndOuter
2. can we make this function as a member function of baseJoiner
3. add a comment for this funtion